### PR TITLE
Td window management

### PR DESCRIPTION
**Pull request by trevordavis**
I temporarily removed the functionality for export, open, and save since it needs refactoring and should be on a separate branch. IPC-middleware is working and is used for changing the canvas color of windows.

### CODE REVIEW

File: electron.js [hunk 1]
```diff
@@ -1,38 +1,251 @@
...
 
-// Keep a global reference of the window object, if you don't, the window will
-// be closed automatically when the JavaScript object is garbage collected.
-let mainWindow;
+// Module to handle window management
+const windowManager = require('electron-window-manager');
+var numWindows = 0;
+var windowNameStart = 'Window #';
+
+const template = [
```
Comment (collinmmckinney) [inline electron.js]: Where did all this code come from / what does it do? Do we need this?

File: src/components/drawing/layers/BackgroundLayerContainer.js [hunk 2]
```diff
@@ -1,15 +1,32 @@
 import { connect } from 'react-redux';
 import BackgroundLayer from './BackgroundLayer';
 
+// const {remote} = window.require('electron');
```
Comment (collinmmckinney) [inline src/components/drawing/layers/BackgroundLayerContainer.js]: Can we remove these commented lines?

File: src/components/ipc-middle/IpcMiddlewareContainer.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { connect } from 'react-redux';
+// import { fileSave } from './../../actions/menu';
+import { canvasColorChange } from './../../actions/canvas';
+import IpcMiddleware from './IpcMiddleware';
+
+const mapStateToProps = (props) => {
+    console.log(props);
```
Comment (collinmmckinney) [inline src/components/ipc-middle/IpcMiddlewareContainer.js]: Try not to leave any console statements in the code, it slows things down when debugging

File: electron.js (hunk 1 above)
Comment (trevordavis) [reply electron.js]: numWindows is incremented each time a new window is created, which is then used with windowNameStart to create a unique window name for each BrowserWindow instance, which is what allows us to create and manage separate browser windows and have actions apply to only specific windows. The template is used for top menu options. I think it should be left in, though I'm planning a refactor in the next week or so.

File: src/components/ipc-middle/IpcMiddlewareContainer.js (hunk 3 above)
Comment (trevordavis) [reply src/components/ipc-middle/IpcMiddlewareContainer.js]: Gone now!

**Pull request closed without merging**